docs/packages/cookbook/how-to-extract-outline-from-markdown.rst

```diff
@@ -0,0 +1,102 @@
+Extract an Outline from a Markdown File
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Accept a Markdown File from the remote user, convert it to text blocks using the markdown blockifier plugin, and then
+generate an outline for the text using the tags.  For example, the following Markdown file:
+
+.. code-block:: markdown
+
+    # This is an h1
+
+    And some content
+
+    ## This is an h2
+
+    and some more content
+
+would generate the following output:
+
+.. code-block:: text
+
+    This is an h1
+        This is an h2
+
+Implementation:
+
+.. code-block:: python
+
+    """
+    This package accepts a Markdown file, extracts its text blocks
+    with a Blockifier, and returns the outline of the content.
+    """
+    from typing import Type
+
+    from steamship import File, MimeTypes
+    from steamship.data import TagKind, DocTag
+    from steamship.invocable import Config, create_handler, post, PackageService
+
+
+    class MarkdownOutlinePackage(PackageService):
+
+        # The config method allows your package to return a class
+        # that defines its required configuration.
+        # See <link> for more details.  This package doesn't have any specific
+        # required configuration, so we return the default Config object.
+        def config_cls(self) -> Type[Config]:
+            """Return Config if your package requires no config."""
+            return Config
+
+        # This method defines the package user's endpoint. The @post annotation
+        # automatically makes the method available as an HTTP Post request.
+        # The name in the annotation defines the HTTP route suffix, see <link>
+        @post("create_markdown_outline")
+        def create_markdown_outline(self, content: str = None) -> str:
+            """Accept markdown content and extract its outline"""
+
+            # Upload the content of the Markdown file into Steamship.
+            file = File.create(self.client, content, mime_type=MimeTypes.MKD)
+
+            # Now blockify it (convert it to raw text with tags) with the markdown blockifier
+            blockifier = self.client.use_plugin("markdown-blockifier-default")
```

Comment:
would it be too much to mention why `instance_handle` isn't used here?

Reply:
Rather than doing that, probably should add it